### ISSUE TITLE
ev(integrate,numer) gives strange result

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3299]
Imported from SourceForge on 2024-07-07 19:56:09
Created by **[dwondrousch](https://sourceforge.net/u/dwondrousch/)** on 2013-09-12 21:14:00
Original: https://sourceforge.net/p/maxima/bugs/2633

---

The following term gives a strange result instead of 0. Without "numer", all is OK
Numerical inaccuracies are a nuisance, but this "false" kills further numerical computations.

~~~~~~~~
ev(integrate(sin(2*%pi*x),x,0,1),numer);
> 0.1591549430919 (1 - cos(7.8398838568245917E-8 (25510582 %pi false - 80143857)))
~~~~~~~~

Info: 
- tested with Maxima 5.30.0 using Windows 7
- tested with Maxima 5.25.0 using Linux Fedora Core 16
- always reproducible
- ev(integrate(sin(x),x,0,2*%pi),numer); computes 0 as expected.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 19:56:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2013-09-13 05:20:22
Original: https://sourceforge.net/p/maxima/bugs/2633/#6b06

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 19:56:14
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2013-09-13 05:20:22
Original: https://sourceforge.net/p/maxima/bugs/2633/#f22b

---

Presence of 'false' in result is caused by failure of PRETTY-GOOD-FLOOR-OR-CEILING to compute floor of 1.0. Avoid the problem by calling $FLOOR instead. Fixed by commit 807d4780. Therefore closing this report.

Note that 807d4780 just removes 'false' from the output -- it is still a messy, although valid, numerical expression.

~~~~
foo : ev(integrate(sin(2*%pi*x),x,0,1),numer)
 => .1591549430918953*(1-cos(7.839883856824592E-8*(25510582*%pi-80143857)))
float (foo);
 => 0.0
~~~~
